include/swift/AST/Identifier.h

```diff
@@ -233,6 +233,12 @@ class DeclBaseName {
     return !isSpecial() && getIdentifier().isEditorPlaceholder();
   }
 
+  /// A representation of the name to be displayed to users. May be ambiguous
+  /// between identifiers and special names
+  StringRef userFacingStr() const {
```

Comment:
Nitpick: I feel like the abbreviation of "str" doesn't make sense in a longer name. How about "userFacingName" or "nameForDiagnostics" or something like that?